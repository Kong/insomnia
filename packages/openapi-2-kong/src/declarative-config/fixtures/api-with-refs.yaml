openapi: 3.0.0

## NOTE: NOT COMPATIBLE WITH KONG GATEWAY >= 3.0 #####
info:
  description: Learn service
  version: 1.0.0
  title: Learn Service

servers:
- url: https://konghq.com/kongu/api/v1/learn
  description: Production server

x-kong-name: awesome_learnservice

paths:
  "/tracks":
    post:
      tags:
      - learn
      summary: Upsert tracks
      operationId: upsertTracks
      x-kong-plugin-request-validator:
        config:
          verbose_response: true
      x-kong-plugin-file-log:
        "$ref": "#/components/x-kong/plugins/log_to_file"
      requestBody:
        "$ref": "#/components/requestBodies/tracks"
      responses:
        '200':
          description: Successful operation

components:
  x-kong:
    # reusable Kong plugin configurations
    plugins:
      log_to_file:
        config:
          path: "/dev/stderr"

  requestBodies:
    tracks:
      content:
        application/json:
          schema:
            type: array
            minItems: 1
            items:
              "$ref": "#/components/schemas/Track"

  schemas:
    Track:
      type: object
      description: Track
      properties:
        id:
          "$ref": "#/components/schemas/TrackId"
        index:
          type: integer
          format: int32
          minimum: 1
          maximum: 100000000
        description:
          "$ref": "#/components/schemas/parameterValue"
    TrackId:
      description: Id of a learning center track
      type: string
      minLength: 1
      maxLength: 5
    parameterValue:
      anyOf:
      - "$ref": "#/components/schemas/symbol"
      - "$ref": "#/components/schemas/sanitized_number"
      - "$ref": "#/components/schemas/symbolArray"
    symbol:
      type: string
      maxLength: 4096
    sanitized_number:
      type: integer
      minimum: -1000000000
      maximum: 1000000000
    symbolArray:
      type: array
      items:
        "$ref": "#/components/schemas/symbol"
      minItems: 0
      maxItems: 1000

