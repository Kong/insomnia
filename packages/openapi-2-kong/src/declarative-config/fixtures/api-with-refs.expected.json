{
  "_format_version": "1.1",
  "services": [
    {
      "name": "awesome_learnservice",
      "protocol": "https",
      "host": "konghq.com",
      "port": 443,
      "path": "/kongu/api/v1/learn",
      "plugins": [],
      "routes": [
        {
          "tags": [
            "OAS3_import",
            "OAS3file_api-with-refs.yaml"
          ],
          "name": "awesome_learnservice-upsertTracks",
          "methods": [
            "POST"
          ],
          "paths": [
            "/tracks$"
          ],
          "strip_path": false,
          "plugins": [
            {
              "name": "file-log",
              "config": {
                "path": "/dev/stderr"
              },
              "tags": [
                "OAS3_import",
                "OAS3file_api-with-refs.yaml"
              ]
            },
            {
              "name": "request-validator",
              "config": {
                "version": "draft4",
                "body_schema": "{\"type\":\"array\",\"minItems\":1,\"items\":{\"type\":\"object\",\"description\":\"Track\",\"properties\":{\"id\":{\"description\":\"Id of a learning center track\",\"type\":\"string\",\"minLength\":1,\"maxLength\":5},\"index\":{\"type\":\"integer\",\"format\":\"int32\",\"minimum\":1,\"maximum\":100000000},\"description\":{\"anyOf\":[{\"type\":\"string\",\"maxLength\":4096},{\"type\":\"integer\",\"minimum\":-1000000000,\"maximum\":1000000000},{\"type\":\"array\",\"items\":{\"type\":\"string\",\"maxLength\":4096},\"minItems\":0,\"maxItems\":1000}]}}}}",
                "allowed_content_types": [
                  "application/json"
                ],
                "verbose_response": true
              },
              "tags": [
                "OAS3_import",
                "OAS3file_api-with-refs.yaml"
              ]
            }
          ]
        }
      ],
      "tags": [
        "OAS3_import",
        "OAS3file_api-with-refs.yaml"
      ]
    }
  ]
}
